x-pack/platform/plugins/shared/ml/public/application/components/anomalies_table/build_alert_params_from_anomaly.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { MlAnomaliesTableRecord } from '@kbn/ml-anomaly-utils';
+import { ML_ANOMALY_RESULT_TYPE, getEntityFieldList } from '@kbn/ml-anomaly-utils';
+import { isDefined } from '@kbn/ml-is-defined';
+import type { MlAnomalyDetectionAlertParams } from '../../../../common/types/alerts';
+import { escapeKueryForFieldValuePair } from '../../util/string_utils';
+
+const SEVERITY_THRESHOLD_ADJUSTMENT = 5;
+
+/**
+ * Builds initial alert parameters from an anomaly record.
+ * Pre-populates job ID, severity, result type, and KQL filter based on the anomaly's characteristics.
+ */
+export function buildAlertParamsFromAnomaly(
+  anomaly: MlAnomaliesTableRecord
+): Partial<MlAnomalyDetectionAlertParams> {
+  const kqlParts: string[] = [];
+
+  // Add entity field filters (partition, over, by fields)
+  const entityFields = getEntityFieldList(anomaly.source);
+  entityFields.forEach((field) => {
+    if (field.fieldName && isDefined(field.fieldValue)) {
+      kqlParts.push(escapeKueryForFieldValuePair(field.fieldName, String(field.fieldValue)));
+    }
+  });
+
+  const actualValue = Array.isArray(anomaly.actual) ? anomaly.actual[0] : anomaly.actual;
```

Comment:
nit, does this also need a `anomaly.actual.length` check? or will it always be populated if it is an array?